pkg/i2gw/provider.go

```diff
@@ -55,13 +55,13 @@ type Provider interface {
 
 type CustomResourceReader interface {
```

Comment:
Maybe we should rename this to `CustomResourceFetcher`? Just thinking loud, I am not sure about it.

Reply:
Good idea, thanks for catching that! Updated.

fyi @LiorLieberman